client/src/components/Login.jsx

```diff
@@ -36,34 +43,47 @@ const Login = () => {
       <h2 className="form__desc">
         <span style={{ color: "red" }}>*</span> = required
       </h2>
-      <FormGroup>
-        <div className="form-group col-md-8">
-          <Input
-            className={"required"}
-            type={"text"}
-            title={"Username"}
-            name={"username"}
-            value={username}
-            placeholder={"Username"}
-            handleChange={(e) => setUsername(e.target.value)}
-          />
-        </div>
-        <div className="form-group col-md-8">
-          <Input
-            className={"required"}
-            type={"password"}
-            title={"Password"}
-            name={"password"}
-            value={password}
-            placeholder={"Password"}
-            handleChange={(e) => setPassword(e.target.value)}
-          />
-        </div>
-        <Button action={handleFormSubmit} type={"primary"} title={"Login"} />
-      </FormGroup>
-      {
-        // <span className="center" >No account? <a href="/signup">Sign up</a> now.</span>
-      }
+      <form
+      onSubmit={handleFormSubmit}
+      className="container-fluid"
+      id="login-form"
+      noValidate>
+        <FormGroup>
+          <div className="form-group col-md-8">
+            <Input
+              className={"required"}
+              type={"text"}
+              title={"Username"}
+              name={"username"}
+              value={username}
+              placeholder={"Username"}
+              handleChange={(e) => setUsername(e.target.value)}
+              errors={errors}
+            />
+          </div>
+          <div className="form-group col-md-8">
+            <Input
+              className={"required"}
+              type={"password"}
+              title={"Password"}
+              name={"password"}
+              value={password}
+              placeholder={"Password"}
+              handleChange={(e) => setPassword(e.target.value)}
+              errors={errors}
+            />
+          </div>
+          {errors && errors.detail && (
+            <div className="form__error-message">
+              {errors.detail}
+            </div>
+          )}
+          <div className="form-group col-md-6" align="center">
+            <Button buttonType={"primary"} type={"submit"} title={"Login"} />
+          </div>
+        </FormGroup>
+      </form>
+      {redirect ? <Redirect to="/" /> : null}
```

Comment:
If I'm reading thsi right, and we want to redirect, we don't need to render the Form part, is that right?  If so, maybe we move the redirect logic to the top and if that state is set, render the <Redirect /> part, otherwise, render the <Form>.